stan/math/prim/fun/sd.hpp

```diff
@@ -22,14 +22,19 @@ namespace math {
  * @return Sample variance.
  */
 template <typename T, require_container_t<T>* = nullptr,
-          require_not_vt_var<T>* = nullptr>
-inline return_type_t<T> sd(const T& m) {
+          require_not_st_var<T>* = nullptr>
+inline auto sd(const T& m) {
   using std::sqrt;
-  check_nonzero_size("sd", "m", m);
-  if (m.size() == 1) {
-    return 0.0;
-  }
-  return sqrt(variance(m));
+
+  return apply_vector_unary<T>::reduce(m, [](const auto& x) {
+    check_nonzero_size("sd", "x", x);
```

Comment:
I think these `check_nonzero_size` checks go inside the `apply_vector_unary`.

Reply:
Should they exists? Like I think normally we just return if a matrix is size zero right?

Reply:
I think in this case it makes sense. What is the standard deviation of `{ }`? I could see an argument for 0.0, but also throwing an error seems to make sense too.

Reply:
That's fair